packages/backend/src/controllers/tasks/tasks.controller.ts

```diff
@@ -0,0 +1,295 @@
+  ApiNotFoundResponse,
+  ApiOkResponse,
+  ApiOperation,
+  ApiParam,
+  ApiResponse,
+  ApiTags,
+} from '@nestjs/swagger';
+import { UserStoreService } from '../../db/user/userStore.service';
+import { HouseStoreService } from '../../db/house/houseStore.service';
+import { Auth } from '../../util/auth.decorator';
+import { TaskUtil } from './tasks.util';
+import { User } from '../../util/user.decorator';
+import { DecodedIdToken } from 'firebase-admin/auth';
+import HouseTasksResponseDto from './dto/house-tasks-response.dto';
+import { CreateTaskDto } from './dto/create-task.dto';
+import { TaskStoreService } from '../../db/task/taskStore.service';
+import { CompleteTaskDto } from './dto/complete-task.dto';
+import UpdateHouseTasksDto from './dto/update-house-tasks.dto';
+import { isValidObjectId } from 'mongoose';
+import { TaskResponseDto } from './dto/task-response-dto';
+
+@ApiTags('tasks')
+@Controller('/api/v1/house/tasks')
+@Auth()
+export class TasksController {
+  constructor(
+    private readonly houseStoreService: HouseStoreService,
+    private readonly userStoreService: UserStoreService,
+    private readonly taskStoreService: TaskStoreService,
+    private readonly taskUtil: TaskUtil,
+  ) {}
+
+  @Post()
+  @ApiOperation({ summary: 'create a new task' })
+  @ApiCreatedResponse({
+    description: 'task created successfully',
+  })
+  @ApiBadRequestResponse({ description: 'Invalid request' })
+  async createTask(
+    @Body() createTaskDto: CreateTaskDto,
+    @User() user: DecodedIdToken,
+  ) {
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+    if (!userDoc.house) {
+      throw new HttpException('user is not in a house', HttpStatus.BAD_REQUEST);
+    }
+
+    const house = await this.houseStoreService.findOne(userDoc.house);
+    if (house) {
+      if (createTaskDto.pool.length < 1) {
+        throw new HttpException('pool is empty', HttpStatus.BAD_REQUEST);
+      }
+
+      createTaskDto.assigned = this.taskUtil.selectRandomUser(
+        createTaskDto.pool,
+      );
+      createTaskDto.house = house._id;
+
+      await this.taskStoreService.create(createTaskDto);
+    }
+  }
+
+  @Get()
+  @ApiOperation({ summary: 'get the current tasks from a house' })
+  @ApiOkResponse({
+    description: 'tasks retrieved successfully',
+    type: HouseTasksResponseDto,
+  })
+  @ApiBadRequestResponse({ description: 'user is not in a house' })
+  async getTasksForHouse(
+    @User() user: DecodedIdToken,
+  ): Promise<HouseTasksResponseDto | null> {
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+    if (!userDoc.house) {
+      throw new HttpException('user is not in a house', HttpStatus.BAD_REQUEST);
+    }
+
+    const house = await this.houseStoreService.findOne(userDoc.house);
+    if (house) {
+      const tasks = await this.taskStoreService.findAll();
+
+      const tasksForHouse = tasks.filter((task) =>
+        task.house.equals(house._id),
+      );
```

Comment:
instead of fetching all and filtering is it possible to add something to the tasks service such as a `findAllByHouse`. This  code is duplicated in this function as well

```typescript
  async findAllByHouse(
    houseId: string | Mongoose.Types.ObjectId,
  ): Promise<TaskDocument[]> {
    return this.taskModel
      .find()
      .populate({ path: 'house', match: { _id: houseId } })
      .exec();
  }
```
This might work?

Reply:
It worked. The code has to be duplicated for updating by the recurrence rules, unfortunately.